test/server/cards/19.1-AS01/Neutral/MangroveSafehouse.spec.js

```diff
@@ -0,0 +1,290 @@
+describe('Mangrove Safehouse', function () {
+    integration(function () {
```

Comment:
You should check the game logs once per "mode" to make sure the effect text is what you want it to be

ex: https://github.com/marcsoiferman/ringteki/blob/625ddbc10ddf709eebfe3ba1f7ecf138930e5f42/test/server/cards/19-ELCore/Unicorn/ShinjoArcher.spec.js#L80